dnsutil/util.go

```diff
@@ -0,0 +1,91 @@
+package dnsutil
+
+import "strings"
+
+// AddDomain adds origin to s if s is not already a FQDN.
+func AddDomain(origin, s string) string {
+	// If you want the result to end in a ".", pass an origin that ends in a "."
```

Comment:
I'd rather leave it as is.  Often user-facing code displays FQDNs without the trailing dot.  Being able to include (or not include) the "." at the end of the origin makes it easy for this one function to work in either case.  That's why I didn't call this MakeFqdn() (and AppendIfNotFqdn() is a bit convoluted).